pkg/apply/prune/prune.go

```diff
@@ -146,6 +156,11 @@ func (p *Pruner) Prune(
 		if !opts.DryRunStrategy.ClientOrServerDryRun() {
 			klog.V(4).Infof("deleting object (object: %q)", id)
 			err := p.deleteObject(id, metav1.DeleteOptions{
+				// Only delete the resource if it hasn't already been deleted
+				// and recreated since the last GET. Otherwise error.
+				Preconditions: &metav1.Preconditions{
```

Comment:
Without this precondition another client might delete and re-create the object before we have a chance to delete it. This change will cause that scenario to error with a failed apply. This should be ok because it will succeed on retry. It's kind of like how a client-side apply might fail from conflict if it was changed between GET and PATCH.